File: jmolecules-axon/02-onion-cqrs-es-bank/src/main/java/org/jmolecules/example/axonframework/domain/api/command/transfer/CancelMoneyTransferCommand.java
package org.jmolecules.example.axonframework.domain.api.command.transfer;

import org.jmolecules.architecture.cqrs.annotation.Command;
import org.jmolecules.ddd.annotation.Association;

@Command(namespace = "axon.bank", name = "CancelMoneyTransferCommand")
public record CancelMoneyTransferCommand(

  @Association
  String sourceAccountId,

  String moneyTransferId,

  String reason

) {

}

File: jmolecules-axon/03-onion-axon-cqrs-es-bank/src/main/kotlin/org/jmolecules/example/axonframework/domain/api/command/transfer/CancelMoneyTransferCommand.kt
package org.jmolecules.example.axonframework.domain.api.command.transfer

import org.jmolecules.architecture.cqrs.annotation.Command
import org.jmolecules.ddd.annotation.Association
import org.jmolecules.example.axonframework.domain.api.type.AccountId
import org.jmolecules.example.axonframework.domain.api.type.MoneyTransferId
import org.jmolecules.example.axonframework.domain.api.type.Reason

/**
 * Internal command to cancel money transfer.
 */
@Command(namespace = "axon.bank", name = "CancelMoneyTransferCommand")
data class CancelMoneyTransferCommand (
    @Association
    val sourceAccountId: AccountId,
    val moneyTransferId: MoneyTransferId,
    val targetAccountId: AccountId,
    val reason: Reason,
)
